tests/e2e/e2e_upgrade_handler_test.go

```diff
@@ -1,18 +1,84 @@
 package e2e_test
 
 import (
-	"fmt"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	precisebanktypes "github.com/kava-labs/kava/x/precisebank/types"
 )
 
-// TestUpgradeHandler can be used to run tests post-upgrade. If an upgrade is enabled, all tests
-// are run against the upgraded chain. However, this file is a good place to consolidate all
-// acceptance tests for a given set of upgrade handlers.
-func (suite *IntegrationTestSuite) TestUpgradeHandler() {
+func (suite *IntegrationTestSuite) TestUpgrade_PreciseBankReserveTransfer() {
 	suite.SkipIfUpgradeDisabled()
-	fmt.Println("An upgrade has run!")
-	suite.True(true)
 
-	// Uncomment & use these contexts to compare chain state before & after the upgrade occurs.
-	// beforeUpgradeCtx := suite.Kava.Grpc.CtxAtHeight(suite.UpgradeHeight - 1)
-	// afterUpgradeCtx := suite.Kava.Grpc.CtxAtHeight(suite.UpgradeHeight)
+	beforeUpgradeCtx := suite.Kava.Grpc.CtxAtHeight(suite.UpgradeHeight - 1)
+	afterUpgradeCtx := suite.Kava.Grpc.CtxAtHeight(suite.UpgradeHeight)
+
+	grpcClient := suite.Kava.Grpc
+
+	// -----------------------------
+	// Get initial reserve balances
+	evmutilAddr := "kava1w9vxuke5dz6hyza2j932qgmxltnfxwl78u920k"
+	precisebankAddr := "kava12yfe2jaupmtjruwxsec7hg7er60fhaa4uz7ffl"
+
+	previousEvmutilBalRes, err := grpcClient.Query.Bank.Balance(beforeUpgradeCtx, &banktypes.QueryBalanceRequest{
+		Address: evmutilAddr,
+		Denom:   precisebanktypes.IntegerCoinDenom,
+	})
+	suite.Require().NoError(err)
+	suite.Require().NotNil(previousEvmutilBalRes.Balance)
+	suite.Require().True(
+		previousEvmutilBalRes.Balance.Amount.IsPositive(),
+		"should have reserve balance before upgrade",
+	)
+
+	previousPrecisebankBalRes, err := grpcClient.Query.Bank.Balance(beforeUpgradeCtx, &banktypes.QueryBalanceRequest{
+		Address: precisebankAddr,
+		Denom:   precisebanktypes.IntegerCoinDenom,
+	})
+	suite.Require().NoError(err)
+	suite.Require().NotNil(previousPrecisebankBalRes.Balance)
+	suite.Require().True(
+		previousPrecisebankBalRes.Balance.Amount.IsZero(),
+		"should be empty before upgrade",
+	)
+
+	suite.T().Logf("x/evmutil balances before upgrade: %s", previousEvmutilBalRes.Balance)
+	suite.T().Logf("x/precisebank balances before upgrade: %s", previousPrecisebankBalRes.Balance)
+
+	// -----------------------------
+	// After upgrade
+	// - Check reserve balance transfer
+	// - Check reserve fully backs fractional amounts
```

Comment:
is there a way to confirm the remainder was set properly without a new query? maybe checking total fractional balances == reserve (auto assumes there is a 0 remainder)

could be worth considering adding a query for remainder